cmd/aries-js-worker/aries.js

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// TODO not all browsers support private members of classes
+/* @class Aries provides Aries SSI-agent functions. */
+const Aries = new function() {
+    // TODO synchronize access on this map?
+    this._pending = new Map()
+    this._WORKER = new Worker('aries-worker.js')
+    this._WORKER.onmessage = e => {
+        const result = e.data
+        const cb = this._pending.get(result.id)
+        this._pending.delete(result.id)
+        cb(result)
+    }
+
+    /**
+     * TODO - remove.
+     * Used for testing.
+     * @param text
+     * @returns {Promise<String>}
+     */
+    this.echo = function(text) {
+        return new Promise((resolve, reject) => {
+            setTimeout(_ => reject(new Error("timeout while accepting invitation")), 5000)
+            const msg = this._newMsg("test", "echo", text)
+            this._pending.set(msg.id, result => {
```

Comment:
this bit of code looks like it could be generic.

Reply:
done